src/registry.ts

```diff
@@ -1,25 +1,35 @@
-import { CommentFactory } from './factory';
+import { ACommentFactory, CellCommentFactory, CellSelectionCommentFactory } from './factory';
 
 export interface ICommentRegistry {
-  createFactory: <T>(options: CommentFactory.IOptions<T>) => CommentFactory<T>;
-  getFactory: (id: string) => CommentFactory | undefined;
+  // createFactory: <T>(options: ACommentFactory.IOptions<T>) => ACommentFactory<T>;
+  getFactory: (id: string) => ACommentFactory | undefined;
+  addFactory: (factory: ACommentFactory | CellCommentFactory | CellSelectionCommentFactory) => void;
 
-  readonly factories: Map<string, CommentFactory>;
+  readonly factories: Map<string, ACommentFactory>;
 }
 
 /**
  * A class that manages a map of `CommentFactory`s
  */
 export class CommentRegistry implements ICommentRegistry {
-  createFactory<T>(options: CommentFactory.IOptions<T>): CommentFactory<T> {
-    const factory = new CommentFactory<T>(options);
-    this.factories.set(options.type, factory);
-    return factory;
+  // createFactory<T>(options: ACommentFactory.IOptions<T>): ACommentFactory<T> {
+  //   const factory = new CommentFactory<T>(options);
+  //   this.factories.set(options.type, factory);
+  //   return factory;
+  // }
+
+  addFactory(factory: CellCommentFactory | CellSelectionCommentFactory): void{
+    if (factory instanceof CellCommentFactory){
```

Comment:
Isn't this equivalent to `this.factories.set(factory.type, factory)`?

Reply:
Yeah that's a bit more concise I'll make the change 👍🏾